core/src/main/java/com/netflix/conductor/core/reconciliation/DefaultWorkflowPoller.java

```diff
@@ -10,42 +10,42 @@
  *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
  *  specific language governing permissions and limitations under the License.
  */
-package com.netflix.conductor.core.execution;
+package com.netflix.conductor.core.reconciliation;
+
+import static com.netflix.conductor.core.execution.WorkflowExecutor.DECIDER_QUEUE;
 
 import com.netflix.conductor.core.LifecycleAwareComponent;
 import com.netflix.conductor.core.config.ConductorProperties;
 import com.netflix.conductor.dao.QueueDAO;
 import com.netflix.conductor.metrics.Monitors;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Component;
 
-import java.util.List;
-import java.util.concurrent.CompletableFuture;
-
-import static com.netflix.conductor.core.execution.WorkflowExecutor.DECIDER_QUEUE;
-
 @SuppressWarnings("SpringJavaInjectionPointsAutowiringInspection")
 @Component
-@ConditionalOnProperty(name = "conductor.workflow-sweeper.enabled", havingValue = "true", matchIfMissing = true)
-public class WorkflowPoller extends LifecycleAwareComponent {
+@ConditionalOnProperty(name = "conductor.default-workflow-sweeper.enabled", havingValue = "true", matchIfMissing = true)
+public class DefaultWorkflowPoller extends LifecycleAwareComponent implements WorkflowPoller {
```

Comment:
The `WorkflowPoller` is not called by any of the Conductor internal classes. So, there is not really a need to define a contract that all pollers should follow. Technically, a bespoke poller can be written (without implementing WorkflowPoller) and used just by disabling the `DefaultWorkflowPoller`.

Reply:
good point, changed